app/src/main/java/com/example/weather/CityPickerViewModel.kt

```diff
@@ -1,11 +1,39 @@
 package com.example.weather
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.weather.network.City
+import com.example.weather.network.WeatherApi
+import com.example.weather.network.WeatherApiResponse
+import kotlinx.coroutines.launch
 
 class CityPickerViewModel : ViewModel() {
 
-    private val _cities: List<Int> = (1..100).toList()
-    val cities: List<Int>
-    get() = _cities
+    private val _cities = MutableLiveData<List<City>>()
 
+    val cities: MutableLiveData<List<City>>
+        get() = _cities
+
+    init {
+        getCities()
+    }
+
+    /**
+     * Gets city information from the Weather API Retrofit service and updates the
+     * [cities] [List] [LiveData].
+     */
+
+    private fun getCities() {
+        viewModelScope.launch {
```

Comment:
What is the `viewModelScope`? What does it do for us? Also what does the `launch` mean?